napari/_qt/dialogs/preferences_dialog.py

```diff
@@ -150,7 +150,26 @@ def _get_page_dict(self, field: 'ModelField') -> Tuple[dict, dict]:
         """Provides the schema, set of values for each setting, and the
         properties for each setting."""
         ftype = cast('BaseModel', field.type_)
-        schema = json.loads(ftype.schema_json())
+        if field.name == 'shortcuts':
+            # FIXME hardcode workaround because pydantic's schema generation
+            #       does not allow you to specify custom JSON serialization
```

Comment:
Can you reword this comment a bit? It looks like something you don't intend to fix right now and I want it to be clear when/how to fix it later. Especially if there's a pydantic issue or something to link to.

Reply:
@nclack hmmm, honestly my proposed fix would be to straight up remove all of the schema autogeneration code and have actual concrete classes for our preference views

we have a whole ass library (`qt_json_builder`) vendored just to make some basic widgets and it adds unnecessary code complexity due to it being too "magical", not to mention having less control and ability to customize the dialogs

not sure if that's the right fix / how to communicate that

Reply:
updated comment works! thanks